### PR TITLE
Feature/tab1

### PR DESCRIPTION
**Pull request by apmccartney**

### CODE REVIEW

File: src/njoy21/input/argument/parser/test/parser.test.cpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#define CATCH_CONFIG_MAIN
+
+#include "catch.hpp"
```
Comment (jlconlin) [inline src/njoy21/input/argument/parser/test/parser.test.cpp]: Is there more testing that needs to be done here?

Comment (apmccartney) [inline src/njoy21/input/argument/parser/test/parser.test.cpp]: See the [directory](https://github.com/njoy/NJOY21/tree/feature/TAB1/src/njoy21/input/argument/parser/test)

Comment (jlconlin) [inline src/njoy21/input/argument/parser/test/parser.test.cpp]: Ah. I did see all those tests (for which I was quite excited), but I didn't make the connection.

**Comment by apmccartney** [comment]
So there were a number of hability improvements made as part of the work

**Comment by jlconlin** [comment]
What's not clear to me is what is fundamentally changed. It appears that there is some cleanup and reshuffling work done, but I'm not sure what was added in the process.

**Comment by apmccartney** [comment]
> how does this change the way I write the input checks?

It doesn't. I just allows you to define policies with `Value_t` aliases which refer to types without a default constructor.

**Pull request merged**